Nightly partest fastOpt is broken with 3 Output differs

Seems like some `NoWarnMissing` are not taken into account.
https://scala-webapps.epfl.ch/jenkins/job/scalajs-task-worker/25180/consoleFull

```
!! 360 - run/private-inline.scala                  [output differs]
% diff /localhome/jenkins/a/workspace/scalajs-task-worker_2/partest/fetchedSources/2.11.2/test/files/run/private-inline-run.log /localhome/jenkins/a/workspace/scalajs-task-worker_2/partest/fetchedSources/2.11.2/test/files/run/private-inline.check
@@ -10,126 +10,4 @@ private-inline.scala:29: warning: Could not inline required method wrapper2 beca
 private-inline.scala:29: warning: At the end of the day, could not inline @inline-marked method wrapper2
   def f2b() = identity(wrapper2(5))
                                ^
-Referring to non-existent method jl_Class$.forName__T__jl_Class
-  called from LTest$$anonfun$main$1.apply__T__V
-  called from LTest$$anonfun$main$1.apply__O__O
-  called from sc_TraversableLike$class.$$anonfun$4__p0__sc_TraversableLike__O__scm_Builder__F1__scm_Builder
-  called from sc_TraversableLike$class.flatMap__sc_TraversableLike__F1__scg_CanBuildFrom__O
-  called from sci_Stream.flatMap__F1__scg_CanBuildFrom__O
-  called from sci_Stream$StreamBuilder.result__sci_Stream
-  called from sci_Stream$StreamBuilder.result__O
-  called from sc_TraversableOnce$class.to__sc_TraversableOnce__scg_CanBuildFrom__O
-  called from sc_AbstractIterator.to__scg_CanBuildFrom__O
-  called from sc_TraversableOnce$class.toList__sc_TraversableOnce__sci_List
-  called from sc_AbstractTraversable.toList__sci_List
-  called from sci_List$.apply__sc_Seq__sci_List
-  called from scm_LazyBuilder.$$plus$eq__O__scm_LazyBuilder
-  called from scm_LazyBuilder.$$plus$eq__O__scm_Builder
-  called from sc_IterableLike$class.take__sc_IterableLike__I__O
-  called from sc_AbstractIterable.take__I__O
-  called from scm_ListBuffer.$$plus$plus$eq__sc_TraversableOnce__scm_ListBuffer
-  called from sci_ListSet$ListSetBuilder.init___sci_ListSet
-  called from sci_ListSet$ListSetBuilder.init___
-  called from sci_ListSet$.newBuilder__scm_Builder
-  called from scg_GenericCompanion.empty__sc_GenTraversable
-  called from scg_GenericSetTemplate$class.empty__scg_GenericSetTemplate__sc_GenSet
-  called from scm_AbstractSet.empty__sc_GenSet
-  called from scm_HashSet.empty__sc_Set
-  called from scm_SetLike$class.newBuilder__scm_SetLike__scm_Builder
-  called from scm_AbstractSet.newBuilder__scm_Builder
-  called from sc_IterableLike$class.take__sc_IterableLike__I__O
-    (already seen, not repeating call stack)
-involving instantiated classes:
-  LTest$$anonfun$main$1
-  sci_Stream$Empty$
-  sci_Stream$Cons
-    (already seen, not repeating call stack)
-  sci_Stream$StreamBuilder
-    (already seen, not repeating call stack)
-  sci_HashSet$HashTrieSet$$anon$1
-    (already seen, not repeating call stack)
-  sci_VectorIterator
-    (already seen, not repeating call stack)
-  sc_IndexedSeqLike$Elements
-    (already seen, not repeating call stack)
-  sci_StreamIterator
-    (already seen, not repeating call stack)
-  sr_ScalaRunTime$$anon$1
-    (already seen, not repeating call stack)
-  sc_Iterator$$anon$2
-    (already seen, not repeating call stack)
-  sci_ListSet$$anon$1
-    (already seen, not repeating call stack)
-  scm_FlatHashTable$$anon$1
-    (already seen, not repeating call stack)
-  sc_LinearSeqLike$$anon$1
-    (already seen, not repeating call stack)
-  scm_ListBuffer$$anon$1
-    (already seen, not repeating call stack)
-  scm_WrappedArray$ofShort
-    (already seen, not repeating call stack)
-  scm_WrappedArray$ofBoolean
-    (already seen, not repeating call stack)
-  scm_WrappedArray$ofRef
-    (already seen, not repeating call stack)
-  scm_StringBuilder
-    (already seen, not repeating call stack)
-  scm_WrappedArray$ofInt
-    (already seen, not repeating call stack)
-  scm_WrappedArray$ofByte
-    (already seen, not repeating call stack)
-  sjs_js_WrappedArray
-    (already seen, not repeating call stack)
-  scm_WrappedArray$ofUnit
-    (already seen, not repeating call stack)
-  scm_WrappedArray$ofFloat
-    (already seen, not repeating call stack)
-  scm_WrappedArray$ofChar
-    (already seen, not repeating call stack)
-  scm_WrappedArray$ofLong
-    (already seen, not repeating call stack)
-  scm_WrappedArray$ofDouble
-    (already seen, not repeating call stack)
-  sci_Range
-    (already seen, not repeating call stack)
-  sci_WrappedString
-    (already seen, not repeating call stack)
-  sci_Range$Inclusive
-    (already seen, not repeating call stack)
-  scm_ArrayBuffer
-    (already seen, not repeating call stack)
-  sci_Vector
-    (already seen, not repeating call stack)
-  sci_List$
-    (already seen, not repeating call stack)
-  scm_ListBuffer
-    (already seen, not repeating call stack)
-  sci_ListSet$
-    (already seen, not repeating call stack)
-  sci_Traversable$
-    (already seen, not repeating call stack)
-  sci_Seq$
-    (already seen, not repeating call stack)
-  sci_IndexedSeq$
-    (already seen, not repeating call stack)
-  sc_IndexedSeq$
-    (already seen, not repeating call stack)
-  sjs_js_WrappedArray$
-    (already seen, not repeating call stack)
-  scm_ArrayBuffer$
-    (already seen, not repeating call stack)
-  sc_Traversable$
-    (already seen, not repeating call stack)
-  sc_Iterable$
-    (already seen, not repeating call stack)
-  scm_IndexedSeq$
-    (already seen, not repeating call stack)
-  sc_Seq$
-    (already seen, not repeating call stack)
-  scm_ListBuffer$
-    (already seen, not repeating call stack)
-  sci_Iterable$
-    (already seen, not repeating call stack)
-  scm_HashSet
-    (already seen, not repeating call stack)
 20
```

```
!! 468 - run/value-class-extractor-seq.scala       [output differs]
% diff /localhome/jenkins/a/workspace/scalajs-task-worker_2/partest/fetchedSources/2.11.2/test/files/run/value-class-extractor-seq-run.log /localhome/jenkins/a/workspace/scalajs-task-worker_2/partest/fetchedSources/2.11.2/test/files/run/value-class-extractor-seq.check
@@ -1,13 +1,3 @@
-Referring to non-existent method jl_Class$.forName__T__Z__jl_ClassLoader__jl_Class
-  called from sr_ScalaRunTime$.isSubClassOf$1__p1__jl_Class__T__Z
-  called from sr_ScalaRunTime$.isXmlNode$1__p1__jl_Class__Z
-  called from sr_ScalaRunTime$.useOwnToString$1__p1__O__Z
-  called from sr_ScalaRunTime$.inner$1__p1__O__I__T
-  called from sr_ScalaRunTime$.$$anonfun$6__p1__O__I__T
-  called from sr_ScalaRunTime$.inner$1__p1__O__I__T
-    (already seen, not repeating call stack)
-involving instantiated classes:
-  sr_ScalaRunTime$
 Bip(1, 2, 3)
 Bip(1, 2, c @ Array(3, 4, 5): _*)
 class [I
```

```
!! 702 - run/t3493.scala                           [output differs]
% diff /localhome/jenkins/a/workspace/scalajs-task-worker_2/partest/fetchedSources/2.11.2/test/files/run/t3493-run.log /localhome/jenkins/a/workspace/scalajs-task-worker_2/partest/fetchedSources/2.11.2/test/files/run/t3493.check--- empty
+++ t3493-run.log
@@ -1,0 +1,10 @@
+Referring to non-existent method jl_Class$.forName__T__Z__jl_ClassLoader__jl_Class
+  called from sr_ScalaRunTime$.isSubClassOf$1__p1__jl_Class__T__Z
+  called from sr_ScalaRunTime$.isXmlNode$1__p1__jl_Class__Z
+  called from sr_ScalaRunTime$.useOwnToString$1__p1__O__Z
+  called from sr_ScalaRunTime$.inner$1__p1__O__I__T
+  called from sr_ScalaRunTime$.$$anonfun$6__p1__O__I__T
+  called from sr_ScalaRunTime$.inner$1__p1__O__I__T
+    (already seen, not repeating call stack)
+involving instantiated classes:
+  sr_ScalaRunTime$
```

--- a/tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/Linker.scala
+++ b/tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/Linker.scala
@@ -101,26 +101,22 @@ final class Linker(semantics: Semantics, considerPositions: Boolean) {
         methodNoWarnsBuf += m
     }
 
-    val trueFct = (_: String) => true
-
     val methodNoWarns = for {
       (className, elems) <- methodNoWarnsBuf.groupBy(_.className)
     } yield {
-      if (classNoWarns.contains(className))
-        className -> trueFct
-      else
-        className -> elems.map(_.methodName).toSet
+      className -> elems.map(_.methodName).toSet
     }
 
-    errors filter {
+    errors filterNot {
       case MissingClass(info, _) =>
-        !classNoWarns.contains(info.encodedName)
+        classNoWarns.contains(info.encodedName)
       case MissingMethod(info, _) =>
-        methodNoWarns.get(info.owner.encodedName).fold(true) { setf =>
-          !setf(info.encodedName)
+        classNoWarns.contains(info.owner.encodedName) ||
+        methodNoWarns.get(info.owner.encodedName).exists { setf =>
+          setf(info.encodedName)
         }
       case _ =>
-        true
+        false
     }
   }
 

